State.py

```diff
@@ -1,119 +1,108 @@
 import threading
 import RaftMessages_pb2 as protoc
 
 class State():
-  def __init__(self, termNumber, server):
-    self.termNumber = termNumber  
-    self.server = server
+  def __init__(self):
+    self.termNumber = 0 
+
+  def replyAENACK(self, toAddr, toPort, termNumber):
+    message = protoc.AppendReply()
+    message.toAddr = toAddr
+    message.toPort = toPort
+    message.term = termNumber
+    message.success = False
+    return protoc.APPENDREPLY, message
+
+  def replyAEACK(self, toAddr, toPort, termNumber):
+    message = protoc.AppendReply()
+    message.toAddr = toAddr
+    message.toPort = toPort
+    message.term = termNumber
+    message.success = True
+    return protoc.APPENDREPLY, message
 
   def sendVoteNACK(self, toAddr, toPort, termNumber):
-    print("Send VoteReply NACK to ", end="")
     voteack = protoc.VoteResult()
     voteack.toAddr = toAddr
     voteack.toPort = toPort
     voteack.term = termNumber
     voteack.granted = False
-    self.server.talk(protoc.VOTERESULT, voteack)
+    return protoc.VOTERESULT, voteack
 
   def sendVoteACK(self, toAddr, toPort, termNumber):
-    print("Send VoteResult ACK to ", end="")
     voteack = protoc.VoteResult()
     voteack.toAddr = toAddr
     voteack.toPort = toPort
     voteack.term = termNumber
     voteack.granted = True
-    self.server.talk(protoc.VOTERESULT, voteack)
+    return protoc.VOTERESULT, voteack
 
-  def replyAENACK(self, toAddr, toPort, termNumber):
-    print("Send AppendReply NACK to ", end="")
-    message = protoc.AppendReply()
-    message.toAddr = toAddr
-    message.toPort = toPort
-    message.term = termNumber
-    message.success = False
-    self.server.talk(protoc.APPENDREPLY, message)
-
-  def replyAEACK(self, toAddr, toPort, termNumber):
-    print("Send AppendReply ACK to ", end="")
-    message = protoc.AppendReply()
-    message.toAddr = toAddr
-    message.toPort = toPort
-    message.term = termNumber
-    message.success = True
-    self.server.talk(protoc.APPENDREPLY, message)   
 
 class LeaderState(State):
-  def __init__(self, termNumber, server):
-    State.__init__(self, termNumber, server)
+  def __init__(self):
+    State.__init__(self)
+    # Send Heartbeats once transitioned to inform a leader is present
     self.sendHeartbeat()
-    self.heartbeat = 2  # interval between heartbeat messages (this must be less than election timout lower bound)
-    #self.initTimer()
+    # Time interval to send messages in seconds
+    self.heartbeat = 4
+    self.timer = None
   
   def initTimer(self):
-    self.timer = threading.Timer(4, self.sendHeartbeat)
+    self.timer = threading.Timer(self.heartbeat, self.sendHeartbeat)
     self.timer.start()
 
   def sendHeartbeat(self):
-    #print("Sending HeartBeats")
     message = protoc.AppendEntries()
-    self.server.talk(protoc.APPENDENTRIES, message)
     self.initTimer()
+    return protoc.APPENDENTRIES, message
 
+  # TO-DO
   def handleMessage(self, messageType, message, termNumber):
```

Comment:
So for the handleMessage signature, what is the termNumber supposed to be?  If its the term number stored in the incoming message then that will be stored in the message parameter, if its referring to this node's term number then you can just grab the self parameter right? I'm not sure what this parameter is referring to.